src/commands/settings/index.ts

```diff
@@ -0,0 +1,70 @@
+import { SlashCommandBuilder } from '@discordjs/builders';
+import { CommandInteraction, GuildMember } from 'discord.js';
+import { Command } from '../../utilities/command';
+import { karma } from './karma';
+import { scanMembers } from './scan-members';
+import { setup } from './setup';
+
+// TODO: Not implemented/not validated subcommands:
+// setup: Once we have more than one setting for a guild, we can use setup as a wizard to guide admins through them
+// scan-members: If/when Rusty gets a website, this command will iterate through all existing users to make sure they have a document in firestore
+
+export default class SetupCommand extends Command {
+  async build() {
+    return new SlashCommandBuilder()
+      .setName('settings')
+      .setDescription('Configuration options')
+      .addSubcommand((setup) =>
+        setup.setName('setup').setDescription('First time setup wizard')
+      )
+      .addSubcommand((karma) =>
+        karma
+          .setName('karma')
+          .setDescription(
+            'Set which channels have message reactions tracked for karma'
+          )
+          .addChannelOption((option) =>
+            option
+              .setName('channel')
+              .setDescription('The channel to modify')
+              .setRequired(true)
+          )
+          .addBooleanOption((option) =>
+            option
+              .setName('bool')
+              .setDescription('Enable or disable tracking')
+              .setRequired(false)
+          )
+      );
+  }
+
+  async execute(interaction: CommandInteraction) {
+    const member = interaction.member as GuildMember;
+    const subcommand = interaction.options.getSubcommand();
+
+    // Ensures only admins may use this command
+    if (!member.permissions.has('ADMINISTRATOR')) {
+      interaction.reply(
```

Comment:
Why do we log an error here? All errors are caught one step up so we could throw here and rely on it.

Also, is there a way to lock certain commands to permission level besides this?

Reply:
There is but it'd require overhead that we haven't done yet from what I understand. I can make this throw an error, a lot of this code was already in the repo, just moved around to fit a subcommand structure and I assume this was before we had general error handling